### PR TITLE
Segmentation for Fishermen

### PR DESCRIPTION
**Pull request by giorgioangel**
This high-level, code-free tutorial explains the main steps of the Thaumato Anakalyptor pipeline in an intuitive way.

### CODE REVIEW

File: docs/29_segmentation_tutorial.md [hunk 1]
```diff
@@ -0,0 +1,253 @@
+---
+title: "Segmentation for Fishermen"
```
Comment (danieldegroot2) [inline docs/29_segmentation_tutorial.md]:
```suggestion
title: "Tutorial 5: Segmentation for Fishermen"
```
Keeps header consistent with existing tutorials.

File: docs/29_segmentation_tutorial.md [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+    property="twitter:image"
+    content="https://scrollprize.org/img/social/opengraph.jpg?2024-02-27"
+  />
+</head>
+
+import { TutorialsTop } from '@site/src/components/TutorialsTop';
+
+<TutorialsTop highlightId={5} />
+
+This high-level, code-free tutorial explains the main steps of the `Thaumato Anakalyptor` [^1] pipeline.
```
Comment (danieldegroot2) [inline docs/29_segmentation_tutorial.md]:
```suggestion
This high-level, code-free tutorial explains the main steps of the [`Thaumato Anakalyptor`](https://github.com/schillij95/ThaumatoAnakalyptor) [^1] pipeline.
```
Adds direct link at initial mention of application.
(allowing users to keep reading)

File: docs/29_segmentation_tutorial.md [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+We will tackle the task of segmentation from another perspective: that of a Neapolitan fisherman.
+
+<figure className="text-center mx-auto">
+  <div className="flex flex-wrap justify-center mx-auto">
+    <img className="max-h-[300px] m-2" src="/img/tutorials/fisherman.webp"/>
+  </div>
+  <figcaption className="mt-1">Neapolitan fisherman.</figcaption>
+</figure>
+
+As we saw in the ["Segmentation and Flattening" tutorial](/tutorial2), the ultimate purpose of segmentation is that of identifying a triangular mesh on the wrapped papyrus sheet.
```
Comment (danieldegroot2) [inline docs/29_segmentation_tutorial.md]:
```suggestion
As we saw in the ["Segmentation and Flattening" tutorial](/tutorial3), the ultimate purpose of segmentation is that of identifying a triangular mesh on the wrapped papyrus sheet.
```
Corrects link target based on existing content.

File: docs/29_segmentation_tutorial.md [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+
+Imagine being Neapolitan fishermen, casting our nets not into schools of fish... but onto sheets!
+We want the net to adhere perfectly on the inner part of the sheet, that is next to where ink is supposed to be located. Indeed, the ink lies on the face of the papyrus that looks towards the pole of rotation of the scroll.
+
+Here, adherence means every node of the net should be placed on the inner surface of the sheet, with adjacent nodes also being adjacent on the sheet. Therefore the problem is twofold: *node placement* and *connectivity establishment*.
+
+But how can we cast the net in? After all, the data we have is a 3D volumetric image composed of colored blocks.
+You can think of it as a big volume made of single units of grayscale LEGO blocks (Figure 1, left). The color of a block represents the density of the material in that region of space: clearer means more dense, hence could be papyrus, while darker means less dense, and could be air. The ink is somewhere in there.
```
Comment (danieldegroot2) [inline docs/29_segmentation_tutorial.md]:
```suggestion
You can think of it as a big volume made of single units of grayscale LEGO blocks (Figure 1, left). The color of a block represents the density of the material in that region of space: lighter means more dense, hence could be papyrus, while darker means less dense, and could be air. The ink is somewhere in there.
```
Clarifies wording based on existing content ('darker').

File: docs/29_segmentation_tutorial.md [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+
+<figure className="text-center mx-auto">
+  <div className="flex flex-wrap justify-center mx-auto">
+    <img className="max-h-[250px] m-2" src="/img/tutorials/lego.webp"/>
+    <img className="max-h-[250px] m-2" src="/img/tutorials/scroll_6_7_21.png"/>
+  </div>
+  <figcaption className="mt-1">Figure 1. Scroll data as LEGO blocks and sagittal view of a subvolume slice.</figcaption>
+</figure>
+
+The right image in Figure 1 displays the view of a section of the volumetric image. Imagine to cut the volume in subvolumes. Take one and look at it from above. Every block composing the subvolume (called voxels) will look like a colored square. That is the color of the pixel in the image.
```
Comment (danieldegroot2) [inline docs/29_segmentation_tutorial.md]:
```suggestion
The right image in Figure 1 displays the view of a section of the volumetric image. Imagine cutting the volume in subvolumes. Take one and look at it from above. Every block composing the subvolume (called *voxels*) will look like a colored square. That is the color of the pixel in the image.
```
Corrects grammar.
Changes style for term 'voxels' (used multiple times).

File: docs/29_segmentation_tutorial.md [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+
+# Node placement
```
Comment (danieldegroot2) [inline docs/29_segmentation_tutorial.md]:
```Suggestion
### Node placement
```
Keeps heading style consistent with existing tutorials.
(Also, [h1 headings](https://www.markdownguide.org/basic-syntax/#headings) are quite large for section headings and usually only used for document titles or page headers.)

File: docs/29_segmentation_tutorial.md [hunk 7]
```diff
@@ -0,0 +1,253 @@
...
+To perform *node placement* we have to detect all the blocks composing the inner surface of the sheet, and forget about the rest.
+This accounts to creating a mask where non-edge blocks are flagged with 0s and edge-blocks are flagged with 1s. In layman's terms, color in black all the non edge voxels, and in white the edge voxels.
```
Comment (danieldegroot2) [inline docs/29_segmentation_tutorial.md]:
```suggestion
This accounts to creating a mask where non-edge blocks are flagged with 0s and edge blocks are flagged with 1s. In layman's terms, color in black all the non-edge voxels, and in white the edge voxels.
```
Adds dash between 'adjective' parts.
Removes dash between 'adjective' and noun.

File: docs/29_segmentation_tutorial.md [hunk 8]
```diff
@@ -0,0 +1,253 @@
...
+
+### Surface detection
```
Comment (danieldegroot2) [inline docs/29_segmentation_tutorial.md]:
```Suggestion
#### Surface detection
```

File: docs/29_segmentation_tutorial.md [hunk 9]
```diff
@@ -0,0 +1,253 @@
...
+What `Thaumato Anakalyptor` does is exactly this: it performs surface detection in 3D by convolving the volumetric image with a 3D `Sobel Kernel`.
```
Comment (danieldegroot2) [inline docs/29_segmentation_tutorial.md]:
```suggestion
What `Thaumato Anakalyptor` does is exactly this: it performs surface detection in 3D by convolving the volumetric image with a 3D [`Sobel Kernel`](https://en.wikipedia.org/wiki/Sobel_operator).
```
Adds link for specific term.

File: docs/29_segmentation_tutorial.md [hunk 10]
```diff
@@ -0,0 +1,253 @@
...
+In simpler terms, it computes a discrete approximation of the 3D gradient in color changes. Then, driven by the intuition that the greatest variation in color will be on an edge, it filters out all the voxels for which the magnitude of a gradient is _small_ (below a given threshold).
```
Comment (danieldegroot2) [inline docs/29_segmentation_tutorial.md]:
```suggestion
In simpler terms, it computes a discrete approximation of the 3D color gradient. Then, driven by the intuition that the greatest variation in color will be on an edge, it filters out all the voxels for which the magnitude of a gradient is _small_ (below a given threshold).
```
Clarifies wording based on existing content.

File: docs/29_segmentation_tutorial.md [hunk 11]
```diff
@@ -0,0 +1,253 @@
...
+
+
+<figure className="text-center mx-auto">
+  <div className="flex flex-wrap justify-center mx-auto">
+    <img className="max-h-[300px]" src="/img/tutorials/scroll_mask.png"/>
+</div>
+<figcaption className="mt-1">Figure 2. Sheet inner surface mask for a subvolume (sagittal view).</figcaption>
+</figure>
+
+### Point Cloud Extraction
```
Comment (danieldegroot2) [inline docs/29_segmentation_tutorial.md]:
```Suggestion
#### Point Cloud Extraction
```

File: docs/29_segmentation_tutorial.md [hunk 12]
```diff
@@ -0,0 +1,253 @@
...
+Exploiting the created empty space, our fisherman net can now _enter_ the scroll. Every node of the fisherman's net will be placed on a point in the point cloud. But how exactly? We want the net to overlay a contiguous *patch* of surface, without connecting parallel sheets. This problem will be addressed in the next section.
+
+<figure className="text-center mx-auto">
+  <div className="flex flex-wrap justify-center mx-auto">
+    <img className="max-h-[300px]" src="/img/tutorials/point_cloud.png"/>
+</div>
+<figcaption className="mt-1">Figure 3. Point cloud, colors added for visualization's sake.</figcaption>
+</figure>
+
+# Connectivity establishment
```
Comment (danieldegroot2) [inline docs/29_segmentation_tutorial.md]:
```Suggestion
### Connectivity establishment
```

File: docs/29_segmentation_tutorial.md [hunk 13]
```diff
@@ -0,0 +1,253 @@
...
+We now have a cloud of points, and it's a mess! In order to _fish_ a single, continguous patch of a sheet, we first have to identify adjacent parts of the sheets, dispatching every point in its proper neighborhood.
+
+### Segmentation
```
Comment (danieldegroot2) [inline docs/29_segmentation_tutorial.md]:
```Suggestion
#### Segmentation
```

File: docs/29_segmentation_tutorial.md [hunk 14]
```diff
@@ -0,0 +1,253 @@
...
+
+<figure className="text-center mx-auto">
+  <div className="flex flex-wrap justify-center mx-auto">
+    <img className="max-h-[250px] m-2" src="/img/tutorials/scroll_mask_groups.png"/>
+    <img className="max-h-[250px] m-2" src="/img/tutorials/sheets.png"/>
+  </div>
+  <figcaption className="mt-1">Figure 4. Identifying patches. (Left) Correct and incorrect grouping; (right) identified patches colored in the point cloud.</figcaption>
+</figure>
+
+Thaumato Anakalyptor performs this operation with `Mask3D` [^2], a Deep Neural Network specifically trained for the purpose.
```
Comment (danieldegroot2) [inline docs/29_segmentation_tutorial.md]:
```suggestion
`Thaumato Anakalyptor` performs this operation with `Mask3D` [^2], a Deep Neural Network specifically trained for the purpose.
```
Keeps formatting consistent with existing content.

File: docs/29_segmentation_tutorial.md [hunk 15]
```diff
@@ -0,0 +1,253 @@
...
+The results should look somewhat like the right part of Figure 4 (that represents a different subvolume, less densely packed). The algorithm segments points cloud in contiguous patches of sheet.
```
Comment (danieldegroot2) [inline docs/29_segmentation_tutorial.md]:
```suggestion
The results should look somewhat like the right part of Figure 4 (that represents a different subvolume, less densely packed). The algorithm segments point clouds in contiguous patches of sheet.
```
Changes spelling based on existing content (mixed plural).

File: docs/29_segmentation_tutorial.md [hunk 16]
```diff
@@ -0,0 +1,253 @@
...
+
+Unfortunately, we have to face two unexpected problems:
+1. the algorithm creates patches with _holes_ and in the holes there are other smaller patches;
+2. the areas of the patches is too small.
```
Comment (danieldegroot2) [inline docs/29_segmentation_tutorial.md]:
```suggestion
2. the area of these patches is too small.
```
Clarifies wording based on existing content (mixed plural).

File: docs/29_segmentation_tutorial.md [hunk 17]
```diff
@@ -0,0 +1,253 @@
...
+
+`Thaumato Anakalyptor` tackles this issue with _stitching_.
+
+### Stitching
```
Comment (danieldegroot2) [inline docs/29_segmentation_tutorial.md]:
```Suggestion
#### Stitching
```

File: docs/29_segmentation_tutorial.md [hunk 18]
```diff
@@ -0,0 +1,253 @@
...
+
+<figure className="text-center mx-auto">
+  <div className="flex flex-wrap justify-center mx-auto">
+    <img className="max-h-[250px]" src="/img/tutorials/stitching.png"/>
+</div>
+<figcaption className="mt-1">Figure 5. Stitching adjacent patches (with the same winding number) together.</figcaption>
+</figure>
+
+In Figure 5 different patches have different colors and patches that are stitched together are linked with red arcs. In the end, all the patches that are stitched together will form a segment, hopefully big enough to read on it some columns of text with ink detection models. It is worth saying that the segmentation performed by `Thaumato Anakalyptor` generates _overlapping_ patches. The patch overlap is exploited during the stitching process.
+`Thaumato Anakalyptor` performs stitching in a sort of Monte Carlo fashion. It first builds _an uncertainty graph_ where nodes of this graph are patches and edges are weighted by the amount of the overlap between patches. Many random walks are launched of the graph to build a subgraph (to select a cover of nodes/patches). The covers that selected the edges with the maximal overlap are chosen as final segments.
```
Comment (danieldegroot2) [inline docs/29_segmentation_tutorial.md]:
```suggestion
`Thaumato Anakalyptor` performs stitching in a sort of Monte Carlo fashion. It first builds _an uncertainty graph_ where nodes of this graph are patches and edges are weighted by the amount of overlap between patches. Many random walks are launched of the graph to build a subgraph (to select a cover of nodes/patches). The covers that selected the edges with the maximal overlap are chosen as final segments.
```
Removes redundant 'the'.

File: docs/29_segmentation_tutorial.md [hunk 19]
```diff
@@ -0,0 +1,253 @@
...
+
+### Mesh reconstruction
```
Comment (danieldegroot2) [inline docs/29_segmentation_tutorial.md]:
```Suggestion
#### Mesh reconstruction
```

File: docs/29_segmentation_tutorial.md [hunk 20]
```diff
@@ -0,0 +1,253 @@
...
+<figure className="text-center mx-auto">
+  <div className="flex flex-wrap justify-center mx-auto">
+    <img className="max-h-[250px] m-2" src="/img/tutorials/mesh_0.png"/>
+    <img className="max-h-[250px] m-2" src="/img/tutorials/mesh_1.png"/>
+    <img className="max-h-[250px] m-2" src="/img/tutorials/mesh_2.png"/>
+  </div>
+  <figcaption className="mt-1">Figure 6. Large mesh wrapped several times around the center. (Left) Side view; (center) from above; (right) zoomed-in view.</figcaption>
+</figure>
+
+In Figure 6, the vertices of the mesh are showed in yellow and the edges connecting them in black. It is worth noting from the zoomed-in view that nearby points are connected by edges and form a triangular mesh.
```
Comment (danieldegroot2) [inline docs/29_segmentation_tutorial.md]:
```suggestion
In Figure 6, the vertices of the mesh are shown in yellow and the edges connecting them in black. It is worth noting from the zoomed-in view that nearby points are connected by edges and form a triangular mesh.
```
Changes spelling (present tense).

File: docs/29_segmentation_tutorial.md [hunk 21]
```diff
@@ -0,0 +1,253 @@
...
+
+Congratulations! You automatically fished the scroll, or a subpart of it!
+
+
+
+# Further steps
```
Comment (danieldegroot2) [inline docs/29_segmentation_tutorial.md]:
```Suggestion
### Further steps
```

File: docs/29_segmentation_tutorial.md [hunk 22]
```diff
@@ -0,0 +1,253 @@
...
+In this extra section we describe how from a triangular mesh we can obtain the 2D image of a sheet. This process involves two main steps: _flattening_ and _rendering_.
+
+### Flattening
```
Comment (danieldegroot2) [inline docs/29_segmentation_tutorial.md]:
```Suggestion
#### Flattening
```

File: docs/29_segmentation_tutorial.md [hunk 23]
```diff
@@ -0,0 +1,253 @@
...
+Now that we have a mesh, we must compute a 3D->2D map to obtain a UV (two-dimensional) parametrization for its vertices. This is a fancy way to say that you have to flatten the mesh. After all, our purpose is to eventually read the ink on a _flat_ image.
+
+A community fork [^5], recently merged to `Thaumato Anakalyptor`, implements an algorithm called `SLIM` [^6] that aims to find a map that minimizes an *isometric distortion energy*. This means that all the points that are equidistant in 3D will be mapped as equidistant as possible in 2D.
```
Comment (danieldegroot2) [inline docs/29_segmentation_tutorial.md]:
```suggestion
A community fork [^5], recently merged to `Thaumato Anakalyptor`, implements an algorithm called `SLIM` [^6] that aims to find a map that minimizes the *isometric distortion energy*. This means that all the points that are equidistant in 3D will be mapped as equidistant as possible in 2D.
```
Changes 'an' (one of multiple) to 'the' (singular).

File: docs/29_segmentation_tutorial.md [hunk 24]
```diff
@@ -0,0 +1,253 @@
...
+The vertices of the mesh showed in Figure 6 are displayed using the obtained UV parametrization in Figure 7.
```
Comment (danieldegroot2) [inline docs/29_segmentation_tutorial.md]:
```suggestion
The vertices of the mesh shown in Figure 6 are displayed using the obtained UV parametrization in Figure 7.
```

File: docs/29_segmentation_tutorial.md [hunk 25]
```diff
@@ -0,0 +1,253 @@
...
+<figure className="text-center mx-auto">
+  <div className="flex flex-wrap justify-center mx-auto">
+    <img className="max-h-[500px]" src="/img/tutorials/flat_mesh.png"/>
+  </div>
+  <figcaption className="mt-1">Figure 7. UV parametrization of the vertices of the mesh displayed in Figure 6.</figcaption>
+</figure>
+
+The mesh, along with its UV parametrization, are finally saved in a `.obj` file.
+
+### Rendering
```
Comment (danieldegroot2) [inline docs/29_segmentation_tutorial.md]:
```Suggestion
#### Rendering
```

File: docs/29_segmentation_tutorial.md [hunk 26]
```diff
@@ -0,0 +1,253 @@
...
+It is extremely important that the obtained UV coordinates are still real numbers. This means that between points there is a lot of empty space!
+In order to go back to a "pixel-style" representation, we have to convert the plot in Figure 7 to a pixelized image. Notice that in Figure 7 we displayed axes with ticks, this was not a mistake.
+
+#### Per-pixel-map and layers
```
Comment (danieldegroot2) [inline docs/29_segmentation_tutorial.md]:
```Suggestion
##### Per-pixel-map and layers
```

File: docs/29_segmentation_tutorial.md [hunk 27]
```diff
@@ -0,0 +1,253 @@
...
+For every integer couple (x,y) in the figure we insert a new point. These coordinates (x,y) will be the integers positions (i,j) of pixels in the rendered image. Eventually, we will end up with a number of points way higher than that of the vertices of the mesh. We will identify in which triangle of the triangular mesh (in 2D) this points falls and compute its `barycentric coordinates`: three coefficients that allow to define the position of every point in a triangle as a linear combination of the positions of the vertices. In Figure 8 we show a red point whose position will be `0.2 * vertex1 + 0.3 vertex2 + 0.5 vertex3`.
```
Comment (danieldegroot2) [inline docs/29_segmentation_tutorial.md]:
```suggestion
For every integer couple (x,y) in the figure we insert a new point. These coordinates (x,y) will be the integers' positions (i,j) of pixels in the rendered image. Eventually, we will end up with a number of points way higher than that of the vertices of the mesh. We will identify in which triangle of the triangular mesh (in 2D) these points fall and compute their `barycentric coordinates`: three coefficients that allow to define the position of every point in a triangle as a linear combination of the positions of the vertices. In Figure 8 we show a red point whose position will be `0.2 * vertex1 + 0.3 vertex2 + 0.5 vertex3`.
```
Adds `'` plural possessive punctuation mark.
Changes spelling (mixed plural).

File: docs/29_segmentation_tutorial.md [hunk 28]
```diff
@@ -0,0 +1,253 @@
...
+  <div className="flex flex-wrap justify-center mx-auto">
+    <img className="max-h-[400px] m-2" src="/img/tutorials/barycentric.png"/>
+  </div>
+  <figcaption className="mt-1">Figure 8. A red point in a triangle represented with its barycentric coordinates.</figcaption>
+</figure>
+
+Exploiting the barycentric coordinates, we can then map back every additional point that we inserted in the UV map to its alleged position in 3D space.
+This step is called obtaining a *per-pixel-map* and it's performed in `Thaumato Anakalyptor` using a recently merged community fork [^7].
+
+Why do we need the 3D positions of the new points? Remember that during the first step, *Node Placement*, we totally forgot about the density of the material obtained via the CT scan (its original color) and started working on a black and white mask, that later became a point cloud, and so on. All the colors displayed in Figures 3-7 were only for representation purpose.
```
Comment (danieldegroot2) [inline docs/29_segmentation_tutorial.md]:
```suggestion
Why do we need the 3D positions of the new points? Remember that during the first step, *Node Placement*, we totally forgot about the density of the material obtained via the CT scan (its original color) and started working on a black and white mask, that later became a point cloud, and so on. All the colors displayed in Figures 3-7 were only for representation purposes.
```
Changes spelling (mixed plural).

File: docs/29_segmentation_tutorial.md [hunk 29]
```diff
@@ -0,0 +1,253 @@
...
+
+Now we need to recover that information, and to do so we need to know the 3D position of every new point that is going to become a pixel of our rendered image.
+Unfortunately, the map from integer 2D positions to 3D points will result into points with non integer positions. What does it mean? After all, the original voxels (Figure 1) _only_ have integer positions. What "color" will these new points/pixel then have?
```
Comment (danieldegroot2) [inline docs/29_segmentation_tutorial.md]:
```suggestion
Unfortunately, the map from integer 2D positions to 3D points will result in points with non-integer positions. What does it mean? After all, the original voxels (Figure 1) _only_ have integer positions. What "color" will these new points/pixels then have?
```
Changes spelling (specific combined verb).
Adds dash between adjective parts.
Changes spelling based on existing content (mixed plural).

File: docs/29_segmentation_tutorial.md [hunk 30]
```diff
@@ -0,0 +1,253 @@
...
+
+We are going to compute the color of the new points by `trilinear interpolation` with their closest ones in the volume.
+This will allow us to obtain images with a smooth varying color.
+
+We forget to mention that since ink prediction is performed not on a single layer, but in what is called a surface volume, that is a stack of surfaces parallel to the surface we obtained along this pipeline, we need to store and transform as well the values of the normals to the surface at each point.
+We already had them for the vertices of the mesh, we need to compute as well the ones for the new points using barycentric coordinates.
```
Comment (danieldegroot2) [inline docs/29_segmentation_tutorial.md]:
```suggestion
We already computed them for the vertices of the mesh, so we only need to compute the ones for the new points using barycentric coordinates.
```
Clarifies wording based on existing content.

File: docs/29_segmentation_tutorial.md [hunk 31]
```diff
@@ -0,0 +1,253 @@
...
+
+In Figure 9 we show the ink prediction from the `Phase 1 Grand Prize winning model` [^8] on the obtained rendered segmented from Figure 7.
+
+<figure className="text-center mx-auto">
+  <div className="flex flex-wrap justify-center mx-auto">
+    <img className="max-h-[500px] m-2" src="/img/tutorials/thaumato_ink_detection.png"/>
+  </div>
+  <figcaption className="mt-1">Figure 8. Ink Prediction on the rendered segment from Figure 7.</figcaption>
```
Comment (danieldegroot2) [inline docs/29_segmentation_tutorial.md]:
```suggestion
  <figcaption className="mt-1">Figure 9. Ink Prediction on the rendered segment from Figure 7.</figcaption>
```
Corrects figure numbering based on existing content.
(figure 7 still seems to be the render, figure 8 is different.)

File: docs/29_segmentation_tutorial.md [hunk 32]
```diff
@@ -0,0 +1,253 @@
...
+</figure>
+
+# Additional notes
```
Comment (danieldegroot2) [inline docs/29_segmentation_tutorial.md]:
```Suggestion
### Additional notes
```

File: docs/29_segmentation_tutorial.md [hunk 33]
```diff
@@ -0,0 +1,253 @@
...
+Please notice that every single one of the previously mentioned step can be improved, both in terms of mesh generation quality and computational efficiency.
```
Comment (danieldegroot2) [inline docs/29_segmentation_tutorial.md]:
```suggestion
Please notice that every single one of the previously mentioned steps can be improved, both in terms of mesh generation quality and computational efficiency.
```

File: docs/29_segmentation_tutorial.md [hunk 34]
```diff
@@ -0,0 +1,253 @@
...
+For further information on `Thaumato Anakalyptor` and its roadmap, please read its official [Technical Report](https://github.com/schillij95/ThaumatoAnakalyptor/blob/main/documentation/ThaumatoAnakalyptor___Technical_Report_and_Roadmap.pdf).
+
+
+# References
```
Comment (danieldegroot2) [inline docs/29_segmentation_tutorial.md]:
```Suggestion
### References
```

File: sidebars.js [hunk 35]
```diff
@@ -83,7 +83,7 @@ const sidebars = {
         { type: 'doc', id: 'tutorial2' },
         { type: 'doc', id: 'tutorial3' },
         { type: 'doc', id: 'tutorial4' },
-        // { type: 'doc', id: 'tutorial5' },
+        { type: 'doc', id: 'segmentation_tutorial' },
```
Comment (danieldegroot2) [inline sidebars.js]: Sidenote: there is unused [18_tutorial5.md](https://github.com/ScrollPrize/scrollprize-website/blob/main/docs/18_tutorial5.md) ('Advanced topics').
You can ignore this comment.

File: src/components/TutorialsTop.js [hunk 36]
```diff
@@ -32,6 +32,11 @@ export function TutorialsTop({ highlightId } = {}) {
                 </video>
                 <div className="text-sm">Ink Detection</div>
             </a>
```
Comment (danieldegroot2) [inline src/components/TutorialsTop.js:42]:
```suggestion
            </a>
            <div className="hidden sm:flex mx-2 mb-2 flex-col items-center">
                <div className="relative leading-[150px] py-4 w-[16px] text-center">→</div>
                <div className="text-sm">&nbsp;</div>
            </div>
```
Adds column styling based on existing content.

File: src/components/TutorialsTop.js [hunk 37]
```diff
@@ -32,6 +32,11 @@ export function TutorialsTop({ highlightId } = {}) {
...
+            <a href="/segmentation_tutorial" className={`mb-2 flex flex-col items-center w-[100px] sm:w-[150px] relative box-content p-2 sm:p-4 sm:pb-2 hover:bg-[#fefefe26] rounded-xl ${ highlightId == 5 ? "bg-[#fffefc30] hover:bg-[#fefefe45]" : "" }`}>
+            <img className="max-h-[300px] m-2" src="/img/tutorials/fisherman.webp"/>
+            Segmentation Intuitive Tutorial
```
Comment (danieldegroot2) [inline src/components/TutorialsTop.js]:
```suggestion
            <div className="text-sm">Segmentation: Intuitive Tutorial</div>
```
Adds text styling based on existing content.
Adds colon between topic and subtopic.

File: docs/29_segmentation_tutorial.md [hunk 38]
```diff
@@ -0,0 +1,253 @@
...
+This accounts to creating a mask where non-edge blocks are flagged with 0s and edge-blocks are flagged with 1s. In layman's terms, color in black all the non edge voxels, and in white the edge voxels.
+
+### Surface detection
+What `Thaumato Anakalyptor` does is exactly this: it performs surface detection in 3D by convolving the volumetric image with a 3D `Sobel Kernel`.
+In simpler terms, it computes a discrete approximation of the 3D gradient in color changes. Then, driven by the intuition that the greatest variation in color will be on an edge, it filters out all the voxels for which the magnitude of a gradient is _small_ (below a given threshold).
+
+At the same time, considering for every voxel a vector that points towards the _umbilicus_ of the scroll (the pole of rotation), and taking the dot product between this vector and the gradient computed just before, one can filter out as well all the voxels that are on the face of the sheet that don't look towards the center. In Figure 2 we display a sagittal view of the mask for the subvolume showed in Figure 1. The image was obtained with a modified version of `Thaumato Anakalyptor`.
+
+
```
Comment (danieldegroot2) [inline docs/29_segmentation_tutorial.md]:
```suggestion
```
Removes redundant newline.

File: docs/29_segmentation_tutorial.md [hunk 39]
```diff
@@ -0,0 +1,253 @@
...
+    <img className="max-h-[250px] m-2" src="/img/tutorials/mesh_2.png"/>
+  </div>
+  <figcaption className="mt-1">Figure 6. Large mesh wrapped several times around the center. (Left) Side view; (center) from above; (right) zoomed-in view.</figcaption>
+</figure>
+
+In Figure 6, the vertices of the mesh are showed in yellow and the edges connecting them in black. It is worth noting from the zoomed-in view that nearby points are connected by edges and form a triangular mesh.
+
+Congratulations! You automatically fished the scroll, or a subpart of it!
+
+
```
Comment (danieldegroot2) [inline docs/29_segmentation_tutorial.md]:
```suggestion
```
Removes redundant newline.

File: docs/29_segmentation_tutorial.md [hunk 40]
```diff
@@ -0,0 +1,253 @@
...
+
```
Comment (danieldegroot2) [inline docs/29_segmentation_tutorial.md]:
```suggestion
```
Removes redundant newline.

File: docs/29_segmentation_tutorial.md [hunk 41]
```diff
@@ -0,0 +1,253 @@
...
+    <img className="max-h-[500px] m-2" src="/img/tutorials/thaumato_ink_detection.png"/>
+  </div>
+  <figcaption className="mt-1">Figure 8. Ink Prediction on the rendered segment from Figure 7.</figcaption>
+</figure>
+
+# Additional notes
+Please notice that every single one of the previously mentioned step can be improved, both in terms of mesh generation quality and computational efficiency.
+For further information on `Thaumato Anakalyptor` and its roadmap, please read its official [Technical Report](https://github.com/schillij95/ThaumatoAnakalyptor/blob/main/documentation/ThaumatoAnakalyptor___Technical_Report_and_Roadmap.pdf).
+
+
```
Comment (danieldegroot2) [inline docs/29_segmentation_tutorial.md]:
```suggestion
```
Removes redundant newline.
(remaining ones at end of file are automatically removed.)

File: docs/29_segmentation_tutorial.md [hunk 42]
```diff
@@ -0,0 +1,253 @@
+---
```
Comment (danieldegroot2) [inline docs/29_segmentation_tutorial.md]: You may want to link from the end of [Tutorial 3](https://github.com/ScrollPrize/scrollprize-website/blob/main/docs/06_tutorial3.md) to this tutorial.

**Review by danieldegroot2 — COMMENTED** [review]
Changes spelling, styling. Corrects numbering.

_Note, has numerous hidden(collapsed) comments halfway down._

File: docs/29_segmentation_tutorial.md (hunk 23 above)
Comment (giorgioangel) [reply docs/29_segmentation_tutorial.md]: Actually it is one among many.

File: docs/29_segmentation_tutorial.md [hunk 43]
```diff
@@ -0,0 +1,250 @@
...
+This step is called obtaining a *per-pixel-map* and it's performed in `Thaumato Anakalyptor` using a recently merged community fork [^7].
+
+Why do we need the 3D positions of the new points? Remember that during the first step, *Node Placement*, we totally forgot about the density of the material obtained via the CT scan (its original color) and started working on a black and white mask, that later became a point cloud, and so on. All the colors displayed in Figures 3-7 were only for representation purposes.
+
+Now we need to recover that information, and to do so we need to know the 3D position of every new point that is going to become a pixel of our rendered image.
+Unfortunately, the map from integer 2D positions to 3D points will result in points with non-integer positions. What does it mean? After all, the original voxels (Figure 1) _only_ have integer positions. What "color" will these new points/pixels then have?
+
+We are going to compute the color of the new points by `trilinear interpolation` with their closest ones in the volume.
+This will allow us to obtain images with a smooth varying color.
+
```
Comment (schillij95) [inline docs/29_segmentation_tutorial.md]: Here it would be good to have the raw rendered image of the segment

Comment (giorgioangel) [reply docs/29_segmentation_tutorial.md]: Thanks!

File: docs/29_segmentation_tutorial.md (hunk 42 above)
Comment (giorgioangel) [reply docs/29_segmentation_tutorial.md]: Done!

**Comment by giorgioangel** [comment]
Thanks @danieldegroot2 and @schillij95 for the great suggestions. I committed almost all of them! :)

**Review by stephenrparsons — APPROVED** [review]
Thanks, Giorgio! This is great. I'd love to see the last few images rotated to have correct orientation (I think this may be implemented now?). Also it would be cool to see the actual commands used to do this so people can follow along, though I know part of the design is the no code intuitive explanation. Anyways I'm merging it!

**Pull request merged**